src/contracts/libraries/BeaconChainProofs.sol

```diff
@@ -281,7 +281,14 @@ library BeaconChainProofs {
         require(proofs.timestampProof.length == 32 * (EXECUTION_PAYLOAD_HEADER_FIELD_TREE_HEIGHT),
             "BeaconChainProofs.verifyWithdrawalProofs: timestampProof has incorrect length");
 
-        {
+        if(proofs.proveHistoricalRoot){
+            uint256 historicalBlockHeaderIndex = HISTORICAL_SUMMARIES_INDEX << ((HISTORICAL_SUMMARIES_TREE_HEIGHT + 1) + 1 + (BLOCK_ROOTS_TREE_HEIGHT)) | 
```

Comment:
parentheses here are awkward, can you regroup to `((HISTORICAL_SUMMARIES_TREE_HEIGHT + 1) + (1 + BLOCK_ROOTS_TREE_HEIGHT))`

Reply:
no I think it should be 1 + (BLOCK_ROOTS_TREE_HEIGHT) not (1 + (BLOCK_ROOTS_TREE_HEIGHT))

because there the "1" signifies that extra step of choosing the "block_root_summary" within the individual "historical_summary". Everywhere else it signifies merkelize_with_mixin, where the length of an array is hashed with the root of the array.  I've added a comment to clear this up

Reply:
ah, sounds good. something that would be helpful is literally commending on every single thing being summed what it's supposed to be. maybe we can do this after things are merged into master